### ISSUE TITLE
Should allow for providing credential JSON in different ways

### ISSUE THREAD

**Issue by askdesigners** [google/google-auth-library-nodejs#152]
This library currently cannot work on hosting providers that have ephemeral file systems like Heroku. It would be handy if there were a way to provide this object directly, rather than by file path. 

Since these systems have an ephemeral fs you cannot easily include files there without having them checked into the git repo.

**Comment by ofrobots** [comment]
I believe this should already work:

```js
const GoogleAuth = require('google-auth-library');
const googleAuth = new GoogleAuth();

googleAuth.fromJSON("your JSON key", (err, authClient) => { .... });
```

@askdesigners, @logicoder, @fhuel: does this address your use cases?

I can adjust the docs which don't seem to provide details on these. I would also suggest checking out [google-auto-auth](https://www.npmjs.com/package/google-auto-auth) which provides a higher level API to work with google auth that might be a bit more convenient to use.

**Comment by JustinBeckwith** [comment]
Fixed in #220
